core/src/CMakeLists.txt

```diff
@@ -11,6 +11,11 @@ INSTALL (DIRECTORY
   FILES_MATCHING PATTERN "*.hpp"
 )
 
+INSTALL (DIRECTORY
+  "${CMAKE_CURRENT_SOURCE_DIR}/"
+  DESTINATION ${KOKKOS_HEADER_DIR}
+  FILES_MATCHING PATTERN "*.h"
+)
```

Comment:
I believe you can concatenate multiple `PATTERN` directives
```CMake
INSTALL (DIRECTORY
  "${CMAKE_CURRENT_SOURCE_DIR}/"
  DESTINATION ${KOKKOS_HEADER_DIR}
  FILES_MATCHING
  PATTERN "*.hpp"
  PATTERN "*.h"
 )
```

Reply:
Tested and it works, pushed a commit to implement it